modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/RexUtils.java

```diff
@@ -173,7 +172,7 @@ public static IndexConditions buildSortedIndexConditions(
 
         condition = RexUtil.toCnf(builder(cluster), condition);
 
-        Map<Integer, List<RexCall>> fieldsToPredicates = mapPredicatesToFields(condition, cluster);
+        Int2ObjectOpenHashMap<List<RexCall>> fieldsToPredicates = mapPredicatesToFields(condition, cluster);
```

Comment:
`ImmutableIntList.copyOf(fieldsToPredicates.keySet())` below lead to box/unbox values.
Let's replace with `ImmutableIntList.of(fieldsToPredicates.keySet().toIntArray)`